.evergreen/scripts/compile-libmongocrypt.sh

```diff
@@ -10,13 +10,7 @@ compile_libmongocrypt() {
   # `.evergreen/scripts/kms-divergence-check.sh` to ensure that there is no
   # divergence in the copied files.
 
-  # TODO: once 1.10.0 is released (containing MONGOCRYPT-605) replace the following with:
-  # git clone -q --depth=1 https://github.com/mongodb/libmongocrypt --branch 1.10.0 || return
-  {
-    git clone -q https://github.com/mongodb/libmongocrypt || return
-    # Check out commit containing MONGOCRYPT-614
-    git -C libmongocrypt checkout 9ce5db1ca353a4b82788724257d460a9ce67a3e9
-  }
+  git clone -q --depth=1 https://github.com/mongodb/libmongocrypt --branch 1.10.1 || return
```

Comment:
The task failures appears due to a bug in libmongocrypt. libmongocrypt applies the `BUILD_VERSION` (1.10.0-pre) to the C driver source referenced in `MONGOCRYPT_MONGOC_DIR`. Filed MONGOCRYPT-706. To work around, try specifying the version in the `VERSION_CURRENT` file:

```bash
  git clone -q --depth=1 https://github.com/mongodb/libmongocrypt --branch 1.10.1 || return

  declare -a crypt_cmake_flags=(
    "-DMONGOCRYPT_MONGOC_DIR=${mongoc_dir}"
    "-DBUILD_TESTING=OFF"
    "-DENABLE_ONLINE_TESTS=OFF"
    "-DENABLE_MONGOC=OFF"
  )

  # Specify libmongocrypt version in VERSION_CURRENT, rather than BUILD_VERSION, to work around MONGOCRYPT-706.
  echo "1.10.1" > libmongocrypt/VERSION_CURRENT
```